File: backend/src/main/java/de/bund/digitalservice/ris/caselaw/adapter/database/r2dbc/lookuptable/NormDTO.java
package de.bund.digitalservice.ris.caselaw.adapter.database.r2dbc.lookuptable;

import jakarta.persistence.Id;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.data.relational.core.mapping.Table;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Table("lookuptable_field_of_law_norm")
public class NormDTO {

  public static final NormDTO EMPTY = new NormDTO();

  @Id Long id;
  Long fieldOfLawId;
  String abbreviation;
  String singleNormDescription;
}

File: backend/src/main/kotlin/de/bund/digitalservice/ris/norms/framework/adapter/output/database/dto/NormDto.kt
package de.bund.digitalservice.ris.norms.framework.adapter.output.database.dto

import org.springframework.data.annotation.Id
import org.springframework.data.annotation.Transient
import org.springframework.data.domain.Persistable
import org.springframework.data.relational.core.mapping.Table
import java.util.*

@Table(name = "norms")
data class NormDto(
    @Id
    val guid: UUID,

) : Persistable<UUID> {

    @Transient
    var newEntry: Boolean = true

    override fun getId(): UUID = guid

    override fun isNew(): Boolean = newEntry
}
